core/src/test/scala/unit/kafka/coordinator/TransactionLogManagerTest.scala

```diff
@@ -0,0 +1,103 @@
+/**
+  * Licensed to the Apache Software Foundation (ASF) under one or more
+  * contributor license agreements.  See the NOTICE file distributed with
+  * this work for additional information regarding copyright ownership.
+  * The ASF licenses this file to You under the Apache License, Version 2.0
+  * (the "License"); you may not use this file except in compliance with
+  * the License.  You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package kafka.coordinator
+
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.record.{CompressionType, KafkaRecord, MemoryRecords}
+
+import scala.collection.JavaConverters._
+
+import org.junit.Test
+import org.junit.Assert.{assertEquals, fail}
+
+class TransactionLogManagerTest {
+
+  val epoch: Short = 0
+  val transactionTimeoutMs: Int = 1000
+
+  @Test
+  def shouldReadWritePidMessages() {
+    // generate pid mapping messages
+    val pidMappings = Map[String, Long]("one" -> 1L, "two" -> 2L, "three" -> 3L)
+
+    val pidRecords = pidMappings.map { case (transactionalId, pid) =>
+      val pidMetadata = new PidMetadata(pid, epoch, transactionTimeoutMs)
+      val pidKey = TransactionLogManager.pidKey(transactionalId)
+      val pidValue = TransactionLogManager.pidValue(pidMetadata)
+      new KafkaRecord(pidKey, pidValue)
+    }.toSeq
+
+    val records = MemoryRecords.withRecords(0, CompressionType.NONE, pidRecords: _*)
+
+    for (record <- records.records.asScala) {
```

Comment:
Might be worth asserting the record counts. It ensures that the test can't succeed if, for whatever reason, the record set ends up empty. Same for the test case below.